### PR TITLE
Null safety

### PR DESCRIPTION
**Pull request by sampontec**
Major version changes due to this [documentation](https://dart.dev/null-safety/migration-guide#package-version).

Suggested changes to tests were ignored. Reason for this is to maintain current functionality. Please let me know if this was not the correct way to go about it.

### CODE REVIEW

**Comment by sampontec** [comment]
[Allow Rule Evaluation on Modern Mobile Frameworks](https://trello.com/c/fVz3fNV0/1682-allow-rule-evaluation-on-modern-mobile-frameworks)

**Comment by deckerego** [comment]
@sampontec - how did you convert & validate? Did you use the migration script, or did you identify the problems as part of `dart analyze`?

**Comment by sampontec** [comment]
@deckerego I ran the migration script but reverted suggested changes on the tests which are needed.

**Comment by deckerego** [comment]
> @deckerego I ran the migration script but reverted suggested changes on the tests which are needed.

Please provide some detail of why things were changed and how they were changed. `dart migrate` will provide you an interactive editor with several suggestions, but we don't need to take all of them. As an example, why do we need to indicate `List values` on line 124 is nullable? Would that value ever be null?

File: lib/src/logic_base.dart [hunk 1]
```diff
@@ -121,7 +121,7 @@ class JsonLogic {
 
     // easy syntax for unary operators, like {"var" : "x"} instead of strict {"var" : ["x"]}
     var op = _getOperator(logic);
-    List values = logic[op] is List ? logic[op] : [ logic[op] ];
+    List? values = logic[op] is List ? logic[op] : [ logic[op] ];
```
Comment (sampontec) [inline lib/src/logic_base.dart:124]: This is nullable due to logic being a Map<dynamic, dynamic> which can contain nullable return types. Thus, the output can be null or [null]. However, the bang operators state that we know that there should be a value. If there is an edge case where there is not a value then a runtime error will be generated.

Comment (sampontec) [reply lib/src/logic_base.dart:124]: All the other changes are based off this one change.

Comment (deckerego) [reply lib/src/logic_base.dart:124]: If we removed the `?` operator and the `!` operators from the `apply` function, how could we determine that the function is no longer null safe?

Comment (sampontec) [reply lib/src/logic_base.dart:124]: Check to see if the list is null? E.g. if (values == null) => logic.

**Review by deckerego — CHANGES_REQUESTED** [review]
Can we remove the formatting changes for now?

**Review by deckerego — APPROVED** [review]

**Pull request merged**